base/iterator.jl

```diff
@@ -447,3 +466,59 @@ end
     s, inner, s2 = state
     return done(f.it, s) && done(inner, s2)
 end
+
+
+"""
+    partition(collection, n) -> iterator
+
+Iterate over a collection `n` elements at a time.
+
+```jldoctest
+julia> collect(partition([1,2,3,4,5], 2))
+3-element Array{Array{Int64,1},1}:
+ [1,2]
+ [3,4]
+ [5]
+
+julia> collect(partition("Hello World!", 5))
+3-element Array{ASCIIString,1}:
+ "Hello"
+ " Worl"
+ "d!"
+```
+"""
+function partition{T}(c::T, n::Int)
+    # FIXME Use @require per https://github.com/JuliaLang/julia/pull/15495
+    @assert n > 0
+    return PartitionIterator{T}(c, n)
+end
+
+type PartitionIterator{T}
+    c::T
+    n::Int
+end
+
+eltype{T<:Union{Vector,AbstractString}}(::Type{PartitionIterator{T}}) = T
```

Comment:
Didn't we say it didn't work for strings currently? So better not accept them for now.

Also, it seems to me that for `Vector` this method is equivalent to the general one below, right?